puddleHopping.js

```diff
@@ -0,0 +1,121 @@
+/*
+We have a frog who loves being in the water (especially in puddles). Because of this, he refuses to travel on land. But, he loves hopping from one puddle to the next. 
+So, he's looking to find the area that has the largest number of puddles for him to jump between.
+
+You'll be given an input which is an array of puddles (note that every puddle is a circle), and for each puddle you have 3 pieces of information: 
+x-position of its center, y-position of its center, and its radius. You'll also be given an input jumpSize which tells you how far our frog can jump.
+
+Return the size of the largest group of puddles that our frog can freely jump between.
+*/
+
+const { Queue } = require('./utils/queue');
+
+const getDistance = (nodeOne, nodeTwo) => {
+    const [nodeOneXcoordinate, nodeOneYcoordinate] = nodeOne;
+    const [nodeTwoXcoordinate, nodeTwoYcoordinate] = nodeTwo;
+
+    const xDistance = nodeTwoXcoordinate - nodeOneXcoordinate;
+    const yDistance = nodeTwoYcoordinate - nodeOneYcoordinate;
+
+    const distanceBetweenNodes = Math.sqrt(Math.pow(xDistance, 2) + Math.pow(yDistance, 2));
```

Comment:
This is a great opportunity to write extremely reusable code. Most people would do something like what you did, and that absolutely works.

But, it would be even better if you wrote a generic function that takes in two points and computes the distance between them. That way, future developers wouldn't need to look up the distance formula or anything like that. And if you ever want to change the way you compute distance (say, to Manhattan distance), then it would be a simple change in just one place.